pysolr.py

```diff
@@ -1346,6 +1349,19 @@ def watchAliases(data, stat):
                 self.aliases = None
             LOG.info("Updated aliases: %s", self.aliases)
 
+        def watchClusterDetailState(data, *args, **kwargs):
+            if not data:
+                LOG.warning("No cluster state available: no collections defined?")
+            else:
+                self.collections.update(json.loads(data.decode('utf-8')))
+                LOG.info('Updated collections: %s', self.collections)
+
+        @self.zk.ChildrenWatch(ZooKeeper.COLLECTION_STATUS)
+        def watchClusterState(children):
```

Comment:
This is what's causing the CI failure since it's the same name as the earlier `watchClusterState` method:

https://travis-ci.org/django-haystack/pysolr/jobs/546780831#L370

Since this isn't really the right name, maybe we should rename this?

```suggestion
        def watchCollectionStatus(children):
```

Reply:
You're right I followed Qinwenchao's answer but there was some naming issues, resolved